content/guides/docker-concepts/running-containers/overriding-container-defaults.md

```diff
@@ -0,0 +1,194 @@
+
+```console
+$ docker run -d -p HOST_PORT:CONTAINER_PORT postgres
+```
+
+### Setting environment variables
+
+This option sets an environment variable `foo` inside the container with the value `bar`.
+
+```console
+$ docker run -e foo=bar postgres env
+```
+
+You will see output like the following:
+
+```console
+HOSTNAME=2042f2e6ebe4
+foo=bar
+```
+
+> **Tip**
+>
+> The `.env` file acts as a convenient way to set environment variables for your Docker containers without cluttering your command line with numerous `-e` flags. To use a `.env` file, you can pass `--env-file` option with the `docker run` command.
+> ```console
+> $ docker run --env-file .env postgres env
+> ```
+{ .tip }
+
+### Restricting the container to consume the resources
+
+You can use the `--memory` and `--cpus` flags with the `docker run` command to restrict how much CPU and memory a container can use. For example, you can set a memory limit for the Python API container, preventing it from consuming excessive resources on your host. Here's the command:
+
+```console
+$ docker run -e POSTGRES_PASSWORD=secret --memory="512m" --cpus="0.5" postgres
+ ```
+
+This command limits container memory usage to 512 MB and defines the CPU quota of 0.5 for half a core.
+
+> **Monitor the real-time resource usage**
+>
+> You can use the `docker stats` command to monitor the real-time resource usage of running containers. This helps you understand whether the allocated resources are sufficient or need adjustment.
+{ .tip }
+
+By effectively using these `docker run` flags, you can tailor your containerized application's behavior to fit your specific requirements.
+
+## Try it now
+
+In this hands-on guide, you'll see how to use the `docker run` command to override the container defaults.
+
+1. [Download and install](/get-docker/) Docker Desktop.
+
+### Run multiple instance of the Postgres database
+
+1.  Start a container using the [Postgres image](https://hub.docker.com/_/postgres) with the following command:
+    
+    ```console
+    $ docker run -d -e POSTGRES_PASSWORD=secret -p 5432:5432 postgres
+    ```
+
+    This will start the Postgres database in the background, listening on the standard container port `5432` and mapped to port `5432` on the host machine.
+
+2. Start a new Postgres container mapped to a different port. 
+
+    ```console
+    $ docker run -d -e POSTGRES_PASSWORD=secret -p 5433:5432 postgres
+    ```
+
+    This will start a new Postgres container in the background, listening on the standard container port `5432` but mapped to port `5433` on the host machine. You override the host port just to ensure that this new container doesn't conflict with the existing running container.
+
+3. Verify if both containers are running via the Docker Dashboard.
+
+    ![A screenshot of Docker Dashboard showing the running instances of Postgres containers](images/running-postgres-containers.webp?border=true)
+
+### Run Postgres container in a controlled network
```

Comment:
This section doesn't address the difference between the default bridge and custom bridge networks and DNS resolution. The paragraph seems to suggest that the default bridge is convenient for container intercommunication, but that's only via container ip which isn't very usable. I think this section should demonstrate DNS resolution over a custom bridge instead (not only how to connect to one)

Reply:
Added a tip/section that discusses customer Vs default bridge network.

Reply:
Thanks!